### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by abs0luty**
This pull request resolves some spelling mistakes in the name of identifiers, comments, and changelog 🤗

### CODE REVIEW

File: compiler-core/src/ast.rs [hunk 1]
```diff
@@ -64,12 +64,12 @@ impl TypedModule {
 /// ```
 ///
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct TargettedDefinition {
+pub struct TargetedDefinition {
```
Comment (lpil) [inline compiler-core/src/ast.rs:67]: Please revert this, we use traditional English rather than USA English spelling in this repo.

**Review by lpil — APPROVED** [review]
Thank you!

**Comment by lpil** [comment]
Seems tests have been broken!

**Comment by abs0luty** [comment]
@lpil tests seem to work now

**Comment by lpil** [comment]
The tests are not running, do you know why?

**Comment by lpil** [comment]
Sorry, could you rebase please? Thanks

**Pull request merged**